lib/widgets.js

```diff
@@ -43,6 +44,66 @@ var input = function (type) {
     };
 };
 
+var choiceValueEquals = function (value1, value2) {
+    return !is.array(value1) && !is.array(value2) && String(value1) === String(value2);
+};
+
+var isSelected = function (value, choice) {
+    return value && (is.array(value) ? value.some(choiceValueEquals.bind(null, choice)) : choiceValueEquals(value, choice));
+};
+
+var renderChoices = function (choices, renderer) {
+    return choices.reduce(function (partialRendered, choice) {
+        var isNested = is.array(choice[1]);
+        var renderData = isNested ?
+            { isNested: true, label: choice[0], choices: choice[1] } :
+            { isNested: false, value: choice[0], label: choice[1] };
+        return partialRendered + renderer(renderData);
+    }, '');
+};
+
+var isScalar = function (value) {
+    return !value || is.string(value) || is.number(value) || is.bool(value);
+};
+
+var unifyChoices = function (choices, nestingLevel) {
+    if (nestingLevel < 0) {
+        throw new Error('choices nested too deep');
```

Comment:
let's make this a `RangeError` so it's even easier to locate the error source :-)